permiso/src/main/java/com/greysonparrelli/permiso/PermisoDialogFragment.java

```diff
@@ -48,16 +55,102 @@ public static PermisoDialogFragment newInstance(@Nullable String title, @NonNull
         return dialogFragment;
     }
 
+    public static class Builder {
+        private int titleId = 0;
+        private String title = null;
+        private int buttonTextId = 0;
+        private String message = null;
+        private int messageId = 0;
+        // by default, interpret the dialog msg body as string text
+        private boolean interpretHtml = false;
+        private String buttonText = null;
+
+
+        private Builder() {}
+
+        public Builder(int titleId, int msgBody, int buttonTextId) {
+            this();
+            this.titleId = titleId;
+            this.messageId = msgBody;
+            this.buttonTextId = buttonTextId;
+        }
+
+
+        public int getTitleId() { return titleId; }
+        public String getTitle() { return title; }
+        public String getMessage() { return message; }
+        public int getMessageId() { return messageId;}
+        public int getButtonTextId() { return buttonTextId; }
+        public String getButtonText() { return buttonText; }
+        public boolean isHtml() { return interpretHtml; }
+
+        public Builder setTitleId(int val) { titleId = val; return this; }
+        public Builder setTitle(String val) { title = val; return this; }
+        public Builder setMessageId(String val) { message = val; return this; }
+        public Builder setMessage(int val)    { messageId = val; return this; }
+        public Builder setButtonTextId(int stringId) { buttonTextId = stringId; return this; }
+        public Builder setButtonText(String string) { buttonText = string; return this; }
+        public Builder setHtmlInterpretation(boolean interpretHtml) { this.interpretHtml = interpretHtml; return this; }
+
+
+        public PermisoDialogFragment build() {
+            return PermisoDialogFragment.getInstance(this);
+        }
+
+    }
+
+    private static PermisoDialogFragment getInstance(Builder builder) {
```

Comment:
Can probably have this named ```newInstance()``` so it acts as an overload. Can also move the code so it's located next to the existing ```newInstance()``` method.